robottelo/hosts.py

```diff
@@ -1848,8 +1848,30 @@ def cli(self):
     @contextmanager
     def omit_credentials(self):
         self.omitting_credentials = True
+        # if CLI is already created
+        if self._cli._configured:
+            for file in Path('robottelo/cli/').iterdir():
+                if file.suffix == '.py' and not file.name.startswith('_'):
+                    cli_module = importlib.import_module(f'robottelo.cli.{file.stem}')
+                    for name, obj in cli_module.__dict__.items():
+                        try:
+                            if Base in obj.mro():
+                                getattr(self._cli, name).omitting_credentials = True
+                        except AttributeError:
+                            # not everything has an mro method, we don't care about them
+                            pass
         yield
         self.omitting_credentials = False
```

Comment:
@lhellebr should we check `self._cli._configured` is configured in the teardown phase too? what happens if we don't?

Reply:
I think if someone does
```
with parametrized_enrolled_sat.omit_credentials():
  pass
```
Then CLI won't get instantiated at all (because `parametrized_enrolled_sat.cli`) hasn't been called. I haven't tested that though. In any case, I employed this just as a defensive programming technique to avoid error in cases we may not even think about.